helix-view/src/editor.rs

```diff
@@ -1788,6 +1794,34 @@ impl Editor {
         self.document_by_path(path).map(|doc| doc.id)
     }
 
+    pub fn get_buffer_content(&self, doc_id: DocumentId) -> Option<String> {
+        self.documents.get(&doc_id).map(|doc| doc.text().to_string())
+    }
+
+    pub fn document_by_id(&self, id: DocumentId) -> Option<&Document> {
+        self.documents.get(&id)
+    }
+
+    pub fn get_buffer_content(&self, doc_id: DocumentId) -> Option<String> {
```

Comment:
Some Errors with the file. Perhaps you have forgotten the lint code with `cargo clippy`.